File: app/src/main/java/com/game/tama/command/CommandEat.java
package com.game.tama.command;

import com.game.engine.Time;
import com.game.tama.core.thing.Thing;
import com.game.tama.core.thing.pet.Pet;
import com.game.tama.core.world.World;

public class CommandEat extends Command
{
    private final float totalTime = 1f;
    private float currentTime = 0f;
    private final Thing toEat;

    public CommandEat(final Thing targetToEat)
    {
        this.toEat = targetToEat;
    }

    @Override
    public void start(final Pet pet, final World world)
    {
        super.start(pet, world);
        world.removeThing(toEat);
        pet.children.add(toEat);
    }

    @Override
    protected void doing(final Pet pet, final World world)
    {
        currentTime += Time.deltaTimeS();

        if (currentTime > totalTime)
        {
            pet.children.remove(toEat);
            complete();
        }
    }

    public static Command Eat(final Thing target)
    {
        return CommandFactory.commandWalkAndAdjacentAction(
            target,
            new CommandEat(target));
    }
}

File: app/src/main/java/com/game/tama/command/CommandEat.kt
package com.game.tama.command

import com.game.android.DisplayAdapter
import com.game.tama.core.GameLoop
import com.game.tama.core.World
import com.game.tama.thing.Pet
import com.game.tama.thing.Thing

class CommandEat(targetToEat: Thing) : Command()
{
    private val totalTime: Float = 1f;
    private var currentTime: Float = 0f;
    private val toEat: Thing = targetToEat;

    override fun start(pet: Pet, world: World)
    {
        super.start(pet, world);
        world.removeThing(toEat);
        pet.children.add(toEat);
    }

    override fun doing(pet: Pet, world: World)
    {
        currentTime += GameLoop.deltaTime;

        if (currentTime > totalTime)
        {
            pet.children.remove(toEat);
            complete();
        }
    }

    override fun draw(d: DisplayAdapter)
    {

    }

    companion object
    {
        fun Eat(target: Thing) : Command
        {
            return CommandFactory.commandWalkAndAdjacentAction(target,
                                                               CommandEat(target));
        }
    }
}